### ISSUE TITLE
Fix running mage modules/algorithms under IN_MEMORY_ANALYTICAL mode and streaming

### ISSUE THREAD

**Issue by chrischabot** [memgraph/memgraph#1036]
IN_MEMORY_ANALYTICAL mode should not have transaction locks. However when I run queries that affect a large number of nodes, such as the DETACH DELTE and updating the community_id seen in the screenshots, they error out with "Cannot resolve conflicting transactions"

IMPORTANT:
* have some graph + run some large analytical query + update results in the same query
* concurrently -> have the bluej stream adding data into Memgraph
**-> how to setup the environment -> https://github.com/memgraph/mage/pull/220**

This is on BlueJ where there are constant writes and reads going on, however transaction locks should not be present at all 

![CleanShot 2023-06-20 at 11 19 18@2x](https://github.com/memgraph/memgraph/assets/11455514/17366b98-b24b-41a9-9f87-7faba704e932)
![CleanShot 2023-06-20 at 11 17 22@2x](https://github.com/memgraph/memgraph/assets/11455514/a988c602-1368-43c2-bee3-d69ff9666758)
![CleanShot 2023-06-20 at 11 16 59@2x](https://github.com/memgraph/memgraph/assets/11455514/0311973e-48a2-4103-80e8-1ba140858360)

**Comment by Darych** [comment]
I couldn't reproduce conflicting transactions in IN_MEMORY_ANALYTICAL mode. On the attached screenshot there is IN_MEMORY_TRANSACTIONAL mode.

In IN_MEMORY_ANALYTICAL mode community_detect function sometimes fails with `Invalid ID` error and sometimes crashes. It's related to changes done in parallel. Because module is written in a way assuming that it has always SNAPSHOT isolation level.
I started to work on module changes to make it aware about different modes, so module authors should think about it during module creation.

**Comment by Darych** [comment]
Summary:
1. In #1395 I added `mgp_graph_is_transactional` function to make query modules aware in which mode they are working.
2. In [Mage PR](https://github.com/memgraph/mage/pull/400):
- I added function `GetInnerNodeIdOpt` to support optional return value. This is in future should be renamed to `GetInnerNodeId` and introduce API BREAKING change.
- In `InsertLouvainRecord` fixed core dump if there is no vertex in graph.

Next steps:
1. Rename `GetInnerNodeIdOpt` to `GetInnerNodeId` and fix all query modules to work correctly with new return value.
2. Edit documentation, README and Release notes about this API BREAKING change.
3. Test other query modules in IN_MEMORY_ANALYTICAL mode with parallel changes as described in this issue.
